lib/ansible/modules/crypto/openssl_certificate.py

```diff
@@ -0,0 +1,760 @@
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    privatekey_path: /etc/ssl/private/example.com.pem
+    provider: assertonly
+
+- name: Ensure that the existing certificate is still valid at the winter solstice 2017
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    valid_at: 20171221162800Z
+
+- name: Ensure that the existing certificate is still valid 2 weeks (1209600 seconds) from now
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    valid_in: 1209600
+
+- name: Ensure that the existing certificate is only used for digital signatures and encrypting other keys
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    keyUsage:
+      - digitalSignature
+      - keyEncipherment
+    keyUsage_strict: True
+
+- name: Ensure that the existing certificate can be used for client authentication
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    extendedKeyUsage:
+      - clientAuth
+
+- name: Ensure that the existing certificate can only be used for client authentication and time stamping
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    extendedKeyUsage:
+      - clientAuth
+      - 1.3.6.1.5.5.7.3.8
+    extendedKeyUsage: strict
+
+- name: Ensure that the existing certificate has a certain domain in its subjectAltName
+  openssl_certificate:
+    path: /etc/ssl/crt/example.com.crt
+    provider: assertonly
+    subjectAltName:
+      - www.example.com
+      - test.example.com
+'''
+
+
+RETURN = '''
+filename:
+    description: Path to the generated Certificate
+    returned: changed or success
+    type: string
+    sample: /etc/ssl/crt/www.ansible.com.crt
+'''
+
+
+from random import randint
+import datetime
+import subprocess
+import os
+
+from ansible.module_utils import crypto as crypto_utils
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+try:
+    import OpenSSL
+    from OpenSSL import crypto
+except ImportError:
+    pyopenssl_found = False
+else:
+    pyopenssl_found = True
+
+
+class CertificateError(crypto_utils.OpenSSLObjectError):
+    pass
+
+
+class Certificate(crypto_utils.OpenSSLObject):
+
+    def __init__(self, module):
+        super(Certificate, self).__init__(
+            module.params['path'],
+            module.params['state'],
+            module.params['force'],
+            module.check_mode
+        )
+
+        self.provider = module.params['provider']
+        self.privatekey_path = module.params['privatekey_path']
+        self.privatekey_passphrase = module.params['privatekey_passphrase']
+        self.csr_path = module.params['csr_path']
+        self.cert = None
+        self.privatekey = None
+        self.module = module
+
+    def check(self, module, perms_required=True):
```

Comment:
I didn't see this function being called in the assertonly provider, but I might be mistaken.